bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/shieldtv/ShieldTVMessageParser.java

```diff
@@ -0,0 +1,448 @@
+                        }
+                    } else {
+                        logger.trace("Second Entry");
+                    }
+
+                    // App Name
+
+                    i += 2; // 12 delimiter
+                    st = "" + charArray[i] + "" + charArray[i + 1];
+                    i += 2;
+                    length = Integer.parseInt(st, 16) * 2;
+                    current = i;
+                    for (; i < current + length; i = i + 2) {
+                        st = "" + charArray[i] + "" + charArray[i + 1];
+                        appSBName.append(st);
+                    }
+
+                    // There are times where there is padding here for no reason beyond the specified length.
+                    // Proceed forward until we get to the 22 delimiter
+
+                    st = "" + charArray[i] + "" + charArray[i + 1];
+                    while (!DELIMITER_22.equals(st)) {
+                        i += 2;
+                        st = "" + charArray[i] + "" + charArray[i + 1];
+                    }
+
+                    // App URL
+                    i += 2; // 22 delimiter
+                    st = "" + charArray[i] + "" + charArray[i + 1];
+                    i += 2;
+                    length = Integer.parseInt(st, 16) * 2;
+                    current = i;
+                    for (; i < current + length; i = i + 2) {
+                        st = "" + charArray[i] + "" + charArray[i + 1];
+                        appSBURL.append(st);
+                    }
+                    st = "" + charArray[i] + "" + charArray[i + 1];
+                    if (!DELIMITER_12.equals(st)) {
+                        i += 4; // terminates 2801
+                    }
+                    String appPrepend = appSBPrepend.toString();
+                    String appDN = ShieldTVRequest.encodeMessage(appSBDN.toString());
+                    String appName = ShieldTVRequest.encodeMessage(appSBName.toString());
+                    String appURL = ShieldTVRequest.encodeMessage(appSBURL.toString());
+                    logger.debug("{} - AppPrepend: {} AppDN: {} AppName: {} AppURL: {}", thingId, appPrepend, appDN,
+                            appName, appURL);
+                    appNameDB.put(appDN, appName);
+                    appURLDB.put(appDN, appURL);
+                }
+                if (appCount > 0) {
+                    LinkedHashMap<String, String> sortedAppNameDB = new LinkedHashMap<>();
+                    ArrayList<String> valueList = new ArrayList<>();
+                    for (Map.Entry<String, String> entry : appNameDB.entrySet()) {
+                        valueList.add(entry.getValue());
+                    }
+                    Collections.sort(valueList, new Comparator<String>() {
+                        public int compare(String str, String str1) {
+                            return (str).compareTo(str1);
+                        }
+                    });
```

Comment:
It's the default implementation: 

```suggestion
                    Collections.sort(valueList);
```